library/Vanilla/Theme/ThemeProviderInterface.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+ namespace Vanilla\Theme;
+
+ /**
+  * Interface for providing variables on a theme.
+  */
+interface ThemeProviderInterface {
+    const TYPE_DB = true;
```

Comment:
By the description below I get the feeling that these constants should be called `TYPE_NUMERIC` and `TYPE_ALPHA_NUMERIC`.